src/main/java/de/tum/in/www1/artemis/domain/Feedback.java

```diff
@@ -419,4 +419,9 @@ public double computeTotalScore(double inputScore, Map<Long, Integer> gradingIns
         }
         return totalScore;
     }
+
+    @JsonIgnore
+    public boolean hasDetails() {
+        return !isPositive() || detailText != null;
```

Comment:
This avoids NPEs:
```suggestion
        return Boolean.FALSE.equals(isPositive()) || detailText != null;
```

But now that I'm thinking of it, maybe this might be another possible solution?
```suggestion
        return detailText != null || gradingInstruction != null;
```
(GradingInstructions have their own feedback text and details are available is the feedback has some additional message )

There also is this method in the client:
https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/entities/feedback.model.ts#L115

Reply:
Thanks. I see your point. So is there always a detailText if the feedback is not positive? Then this would be a possibility.

Reply:
Or should I use this instead?
```suggestion
        return !isPositive() || detailText != null || gradingInstruction != null;
```

Reply:
```suggestion
        return !Boolean.TRUE.equals(isPositive()) || detailText != null || gradingInstruction != null;
```
Maybe with one small addition to cover the case of `isPositive() == null`. I think that should then cover all cases.